File: elysia_ai/schemas.py
# =========================================================
# 📍 FILE: elysia_ai/schemas.py
# 📝 ROLE:
#   Input- en outputschemas voor AI-requests.
#   Zorgt voor voorspelbaarheid en veiligheid.
# =========================================================

from pydantic import BaseModel, Field
from typing import Dict


class GenerateRequest(BaseModel):
    input: str
    context: Dict[str, str] = Field(default_factory=dict)


class GenerateResponse(BaseModel):
    success: bool
    output: str
    tokens: int
    error: str | None = None

File: modules/ai/src/schemas.ts
// =========================================================
// 📍 FILE: modules/ai/src/schemas.ts
// 📝 ROLE:
//   Contracts Kernel <-> AI-module
// =========================================================

export type AITask = "generate";

export interface AIExecuteRequest {
  task: AITask;
  input: string;
  context?: Record<string, string>;
}

export interface AIExecuteResponse {
  success: boolean;
  output: string;
  tokens: number;
  error?: string | null;
}
